osf_tests/metrics/reporters/test_institutional_users_reporter.py

```diff
@@ -0,0 +1,257 @@
+
+    def test_one_user_with_stuff_and_no_files(self):
+        self._user_setup_with_stuff.affiliate_user()
+        _reports = list(InstitutionalUsersReporter().report(self._yearmonth))
+        self.assertEqual(len(_reports), 1)
+        self._assert_report_matches_setup(_reports[0], self._user_setup_with_stuff)
+        self.assertEqual(_reports[0].public_file_count, 0)
+        self.assertEqual(_reports[0].storage_byte_count, 0)
+
+    def test_one_user_with_stuff_and_a_file(self):
+        self._user_setup_with_stuff.affiliate_user()
+        _user = self._user_setup_with_stuff.user
+        _project = _user.nodes.first()
+        with _patch_now(self._now):
+            create_test_file(target=_project, user=_user, size=37)
+        (_report,) = InstitutionalUsersReporter().report(self._yearmonth)
+        self._assert_report_matches_setup(_report, self._user_setup_with_stuff)
+        self.assertEqual(_report.public_file_count, 1)
+        self.assertEqual(_report.storage_byte_count, 37)
+
+    def test_one_user_with_stuff_and_multiple_files(self):
+        self._user_setup_with_stuff.affiliate_user()
+        _user = self._user_setup_with_stuff.user
+        _project = _user.nodes.first()
+        with _patch_now(self._now):
+            create_test_file(target=_project, user=_user, size=37, filename='b')
+            create_test_file(target=_project, user=_user, size=73, filename='bl')
+            _component = ProjectFactory(parent=_project, creator=_user, is_public=True)
+            _component.affiliated_institutions.add(self._institution)
+            create_test_file(target=_component, user=_user, size=53, filename='bla')
+            create_test_file(target=_component, user=_user, size=51, filename='blar')
+            create_test_file(target=_component, user=_user, size=47, filename='blarg')
+        (_report,) = InstitutionalUsersReporter().report(self._yearmonth)
+        self._assert_report_matches_setup(_report, self._user_setup_with_stuff)
+        self.assertEqual(_report.public_file_count, 5)
+        self.assertEqual(_report.storage_byte_count, 37 + 73 + 53 + 51 + 47)
+
+    def test_several_users(self):
+        _setups = [
+            self._user_setup_with_nothing,
+            self._user_setup_with_ones,
+            self._user_setup_with_stuff,
+        ]
+        for _setup in _setups:
+            _setup.affiliate_user()
+        _setup_by_userid = {
+            _setup.user._id: _setup
+            for _setup in _setups
+        }
+        _reports = list(InstitutionalUsersReporter().report(self._yearmonth))
+        self.assertEqual(len(_reports), len(_setup_by_userid))
+        for _actual_report in _reports:
+            _setup = _setup_by_userid[_actual_report.user_id]
+            self._assert_report_matches_setup(_actual_report, _setup)
+
+# helper class for test-case setup
+@dataclasses.dataclass
+class _InstiUserSetup:
+    '''oof, so many things to set up, gross'''
```

Comment:
Nit: I'd reiterate any business logic here, like I'd say preprints should be based on `can_view`,  or any opinionated stuff, but NBD.